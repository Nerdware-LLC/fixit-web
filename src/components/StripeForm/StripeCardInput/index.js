export { StripeCardInput } from "./StripeCardInput";
export { useStripeCardInput } from "./useStripeCardInput";
