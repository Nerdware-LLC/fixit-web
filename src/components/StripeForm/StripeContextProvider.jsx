import React from "react";
import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";
import { CONFIG } from "../../config";
import { element } from "../../types";

const stripePromise = loadStripe(CONFIG.STRIPE.PUBLISHABLE_KEY);

export const StripeContextProvider = ({ children }) => (
  <Elements stripe={stripePromise}>{children}</Elements>
);

StripeContextProvider.propTypes = {
  children: element.isRequired
};
