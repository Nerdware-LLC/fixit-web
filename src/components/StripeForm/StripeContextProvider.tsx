import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";
import { ENV } from "@config";
import { element } from "@types";

const stripePromise = loadStripe(ENV.STRIPE.PUBLISHABLE_KEY);

export const StripeContextProvider = ({ children }: { children: React.ReactNode }) => (
  <Elements stripe={stripePromise}>{children}</Elements>
);

StripeContextProvider.propTypes = {
  children: element.isRequired
};
