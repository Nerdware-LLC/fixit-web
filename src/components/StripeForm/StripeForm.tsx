import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";
import { ENV } from "@app/env";
import { StripeFormElements, type StripeFormElementsProps } from "./StripeFormElements";

const stripePromise = loadStripe(ENV.STRIPE_PUBLISHABLE_KEY);

/**
 * A form wrapped in the Stripe Elements provider.
 */
export const StripeForm = (props: StripeFormElementsProps) => (
  <Elements stripe={stripePromise}>
    <StripeFormElements {...props} />
  </Elements>
);

export type StripeFormProps = StripeFormElementsProps;
