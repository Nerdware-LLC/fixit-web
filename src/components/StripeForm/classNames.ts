export const stripeFormClassNames = {
  stripeFormRoot: "stripe-form-root",
  cardInput: {
    container: "stripe-form-card-input-container",
    label: "stripe-form-card-input-label",
    errorMessage: {
      container: "stripe-form-card-input-error-message-container",
      text: "stripe-form-card-input-error-message-text",
    },
  },
  submitButton: "stripe-form-submit-button",
};
