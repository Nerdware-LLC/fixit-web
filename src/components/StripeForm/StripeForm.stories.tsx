import { FetchStateContextProvider } from "@/app/FetchStateContext";
import { authenticatedUserStore } from "@/stores/authenticatedUserStore.js";
import { STATIC_MOCK_USERS } from "@/tests/mockItems/staticMockUsers.js";
import { StripeForm } from "./StripeForm.jsx";
import type { Meta, StoryObj } from "@storybook/react";

const meta = {
  title: "Components/StripeForm",
  component: StripeForm,
  decorators: [
    (Story) => (
      <FetchStateContextProvider>
        <Story />
      </FetchStateContextProvider>
    ),
  ],
} satisfies Meta<typeof StripeForm>;

export default meta;

///////////////////////////////////////////////////////////
// STORIES

type Story = StoryObj<typeof meta>;

authenticatedUserStore.set(STATIC_MOCK_USERS.Guy_McPerson);

export const PaymentInputDemo = {
  args: {
    stripeElementsOptions: {
      mode: "subscription",
      currency: "usd",
      amount: 1000,
    },
  },
} satisfies Story;
