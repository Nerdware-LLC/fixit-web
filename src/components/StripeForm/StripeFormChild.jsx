import React, { useState } from "react";
import styled from "styled-components";
import { useStripeCardInput } from "./StripeCardInput";
import { StripeFormSubmitButton } from "./StripeFormSubmitButton";
import { useFetchStateContext } from "../Indicators";
import { func, string } from "../../types";

export const StripeFormChild = ({ handleSubmit, ...otherProps }) => {
  const { StripeCardInput, getPaymentMethod } = useStripeCardInput();
  const { setIsLoading, setError } = useFetchStateContext();
  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);

  const _handleSubmit = e => {
    e.preventDefault();
    setIsLoading(true);
    setIsSubmitDisabled(true);
    (async () => {
      try {
        const paymentMethod = await getPaymentMethod();
        await handleSubmit(paymentMethod);
      } catch (error) {
        setError(error);
        setIsSubmitDisabled(true);
      } finally {
        setIsLoading(false);
      }
    })();
  };

  return (
    <StyledForm onSubmit={_handleSubmit} {...otherProps}>
      <StripeCardInput setIsSubmitDisabled={setIsSubmitDisabled} />
      <StripeFormSubmitButton isSubmitDisabled={isSubmitDisabled} />
    </StyledForm>
  );
};

const StyledForm = styled.form`
  box-sizing: border-box;
  width: 100%;
  display: flex;
  justify-self: flex-end;
`;

export const STRIPE_FORM_PROP_TYPES = {
  handleSubmit: func.isRequired,
  className: string
};

StripeFormChild.propTypes = STRIPE_FORM_PROP_TYPES;
