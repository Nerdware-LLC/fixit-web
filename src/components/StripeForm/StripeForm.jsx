import React from "react";
import { StripeContextProvider } from "./StripeContextProvider";
import { StripeFormChild, STRIPE_FORM_PROP_TYPES } from "./StripeFormChild";
import { FetchStateContextWrapper } from "../Indicators";

export const StripeForm = ({ handleSubmit, ...otherProps }) => (
  <StripeContextProvider>
    <FetchStateContextWrapper>
      <StripeFormChild handleSubmit={handleSubmit} {...otherProps} />
    </FetchStateContextWrapper>
  </StripeContextProvider>
);

StripeForm.propTypes = STRIPE_FORM_PROP_TYPES;
