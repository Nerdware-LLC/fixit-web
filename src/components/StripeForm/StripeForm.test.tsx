import renderer from "react-test-renderer";
import { StripeForm } from "./StripeForm";

jest.mock("./StripeFormChild", () => ({
  __esModule: true,
  StripeFormChild: () => <span>Mock_StripeFormChild</span>,
}));

it("renders correctly", () => {
  const tree = renderer.create(<StripeForm handleSubmit={jest.fn()} />).toJSON();

  expect(tree).toMatchSnapshot();
});
