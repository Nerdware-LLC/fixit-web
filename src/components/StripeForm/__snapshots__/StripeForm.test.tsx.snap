// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<span>
  Mock_StripeFormElements
</span>
`;
